File: ShapesPainter/app/src/main/java/com/example/alexander/shapespainter/PictureDraft.java
package com.example.alexander.shapespainter;


import com.example.alexander.shapespainter.model.Shape;

import java.util.ArrayList;


public class PictureDraft {
    private ArrayList<Shape> shapeArrayList = new ArrayList<>();

    public void addShape(Shape shape) {
        shapeArrayList.add(shape);
    }

    public int getShapeCount() {
        return shapeArrayList.size();
    }

    public Shape getShape(int idx) {
        return shapeArrayList.get(idx);
    }


    public ArrayList<Shape> getShapes() {
        return shapeArrayList;
    }

    public void deleteAllShapes() {
        shapeArrayList.clear();
    }

    public void removeShape(Shape shape) {
        shapeArrayList.remove(shape);
    }

}

File: Fabric/app/src/main/java/com/example/alexander/fabric/WorkWithShapes/PictureDraft.kt
package com.example.alexander.fabric.WorkWithShapes
import com.example.alexander.fabric.Interfaces.Shape
import java.util.*

/**
 * Created by Alexander on 25.06.2016.
 */
class PictureDraft {
    val shapeList = ArrayList<Shape>()

    fun AddShape(shape: Shape)
    {
        shapeList.add(shape)
        //var x = Pair(1,2)
    }

    fun GetShapeCount(): Int
    {
        return shapeList.size
    }

    fun GetShape(idx: Int): Shape
    {
        return shapeList[idx]
    }
}